src/dp.rs

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use crate::field::FieldElement;
+use crate::vdaf::prg::Seed;
+use std::fmt::Debug;
+
+// TODO(tholop): We could also use pairs of integers, or even
+// floats to store the DiscreteGaussian DP parameters?
```

Comment:
Pairs of integers might be fine. I'm not sure we'd want to use floats: If we end up standardizing some algorithm we're going to need test vectors, and the precision of floating points might cause some issues.

One last thought here: It sounds like we're going to need `BigInt` in any case, based on our discussion about how machine representations of integers might invalidate existing DP performance analysis. Although the additional dependency on `BigRational` is pretty small potatoes, it might be prudent to avoid exposing it in the public API. We could alternatively define our own wrapper type:

```rust
pub struct Rational(BigRational);
```

Thoughts @ooovi? @tgeoghegan? @divergentdave? @branlwyd?

Reply:
Sorry, "DP performance analysis" is the term I learned from @tholop to describe theorems about privacy versus utility. By performance we don't mean CPU time or whatever.

Reply:
I think `Rational::new(numerator: u32, denominator: u32)` would be a reasonable. (1) these numbers would be reasonably short and (2) we want them to be compatible with the field that's used. (`u32` is smaller than every `F::Integer` we have in this repo.)

Reply:
I used `pub struct Rational {numerator: u32, denominator: u32}` for now, to keep it simple. But this is not really part of the API anyway; we might have to change it when we do the actual discrete Gaussian implementation.